### ISSUE TITLE
[Extensions] Unexpected filenames proposed when saving untitled files

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#485]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.66.2
- OS Version: Windows 10

TL/DR: When saving untitled documents that have name with extension, the proposed file name is not the same and can be completely different:
`FooBar.component.yml` -> `.condarc.txt` (I do not have a file with that name ANYWHERE on my PC. I have `.condarc` somewhere though)
`FooBar.component.yaml` -> `FooBar.component.yaml.yml`

Steps to Reproduce:

Issue 1:

1. Create extension command that executes `vscode.commands.executeCommand('vscode.openWith',         vscode.Uri.parse("untitled:FooBar.component.yaml"), "default" );`
2. Debug the extension and activate the command to get a new untitled file tab
3. Try to save the file. The proposed filename is `FooBar.component.yaml.yml` despite the original Untitled file name already ending with `.yaml` which is a valid extension for YAML files

Issue 2:

1. Create extension command that executes `vscode.commands.executeCommand('vscode.openWith',         vscode.Uri.parse("untitled:FooBar.component.yml"), "default" );`
2. Debug the extension and activate the command to get a new untitled file tab
3. Try to save the file. The proposed filename is `.condarc.txt`, which is unexpected and very surprising. I do not have a file with that name **anywhere** on my PC. I have `.condarc` somewhere though. Note that the VSCode-added extension is `.txt` here despite VSCode detecting the file format as YAML based on contents.

I expect that when saving the file VSCode proposes my given filename and extension to the user.
I expect that when saving the file VSCode does not propose completely unrelated file names like `.condarc.txt`.

**Comment by vscodenpa** [comment]
confidence for label extensions: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for assignee bpasero: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for label extensions: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for assignee bpasero: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for label extensions: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for assignee bpasero: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for label extensions: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for assignee bpasero: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for label extensions: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for assignee bpasero: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for label extensions: 0.6. does not meet threshold

**Comment by vscodenpa** [comment]
confidence for assignee bpasero: 0.6. does not meet threshold
